code/modules/clothing/spacesuits/nanosuit.dm

```diff
@@ -0,0 +1,1269 @@
+		visible_message("<span class='danger'>[user] has hit [src] with a strengthened blow from [I]!</span>", null, null, COMBAT_MESSAGE_RANGE)
+		take_damage(I.force*1.75, I.damtype, "melee", TRUE)//take 75% more damage with strength on
+		return
+	return ..()
+
+/obj/item/throw_at(atom/target, range, speed, mob/thrower, spin = TRUE, diagonals_first = FALSE, datum/callback/callback)
+	if(thrower && ishuman(thrower))
+		var/mob/living/carbon/human/H = thrower
+		if(istype(H.wear_suit, /obj/item/clothing/suit/space/hardsuit/nano))
+			var/obj/item/clothing/suit/space/hardsuit/nano/NS = H.wear_suit
+			if(NS.mode == NANO_STRENGTH)
+				.=..(target, range*1.5, speed*2, thrower, spin, diagonals_first, callback)
+				return
+	. = ..()
+
+/datum/martial_art/nanosuit/proc/on_attack_hand(mob/living/carbon/human/owner, atom/target, proximity)
+	if(proximity)
+		return target.attack_nanosuit(owner)
+
+/mob/living/carbon/human/UnarmedAttack(atom/A, proximity)
+	var/datum/martial_art/nanosuit/style = mind?.has_martialart(MARTIALART_NANOSUIT)
+	if(style)
+		if(style.on_attack_hand(src, A, proximity))
+			return
+		else if(iscarbon(A) && !ishuman(A) && style.harm_act(src, A))
+			return
+	..()
+
+/mob/living/simple_animal/attack_hand(mob/living/carbon/human/M)
+	. = ..()
+	if(M && ishuman(M) && istype(M.wear_suit, /obj/item/clothing/suit/space/hardsuit/nano))
+		var/obj/item/clothing/suit/space/hardsuit/nano/NS = M.wear_suit
+		NS.kill_cloak()
+
+/obj/item/clothing/suit/space/hardsuit/nano/proc/kill_cloak()
+	if(mode == NANO_CLOAK)
+		var/obj/item/W = Wearer.get_active_held_item()
+		if(istype(W, /obj/item/gun))
+			var/obj/item/gun/G = W
+			if(G.suppressed && G.can_shoot())
+				cut_nano_energy(15)
+				Wearer.filters = null
+				animate(Wearer, alpha = 255, time = stealth_cloak_out)
+				addtimer(CALLBACK(src, .proc/resume_cloak),6,TIMER_UNIQUE|TIMER_OVERRIDE)
+				return
+		cut_nano_energy(cell.charge,NANO_CHARGE_DELAY)
+
+/obj/item/clothing/suit/space/hardsuit/nano/proc/resume_cloak()
+	if(cell.charge && mode == NANO_CLOAK)
+		Wearer.filters = filter(type="blur",size=1)
+		animate(Wearer, alpha = 40, time = stealth_cloak_in)
+
+/obj/item/storage/box/syndie_kit/nanosuit
+	name = "\improper Crynet Systems kit"
+	desc = "Maximum Death."
+
+/obj/item/storage/box/syndie_kit/nanosuit/PopulateContents()
+	new /obj/item/clothing/suit/space/hardsuit/nano(src)
+
+/obj/item/implant/explosive/disintegrate
+	name = "disintegration implant"
+	desc = "Ashes to ashes."
+	icon_state = "explosive"
+	resistance_flags = INDESTRUCTIBLE | FIRE_PROOF | ACID_PROOF | FREEZE_PROOF
+
+/obj/item/implant/explosive/disintegrate/activate(cause)
+	if(!cause || !imp_in || cause == "emp" || active)
+		return FALSE
+	if(cause == "action_button" && !popup)
+		popup = TRUE
+		var/response = alert(imp_in, "Are you sure you want to activate your [name]? This will cause you to disintergrate!", "[name] Confirmation", "Yes", "No")
```

Comment:
Disintergrate

Reply:
There is a suicide button like the one nuke ops get but this one makes you desintegrate.

Reply:
No, I meant it's mispelled in this line

Reply:
oh yeah it is will fix it reaal soon.